pkg/kubelet/cm/dra/plugin/plugin.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	utilversion "k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// DRAPluginName is the name of the in-tree DRA Plugin.
+	DRAPluginName = "kubernetes.io/dra"
+)
+
+// draPlugins map keeps track of all registered DRA plugins on the node
+// and their corresponding sockets.
+var draPlugins = &PluginsStore{}
+
+// RegistrationHandler is the handler which is fed to the pluginwatcher API.
+type RegistrationHandler struct{}
+
+var PluginHandler = &RegistrationHandler{}
```

Comment:
I like the idea of the `PluginsStore` abstraction, but I'm not a fan of this being held in a global variable like this.
Likewise with the `PluginHandler` global. In my opinion, we should try and avoid the use of these global variables like this as much as possible.

That said, I see that there is precedence for this in the CSI world in e.g. `pkg/volume/csi/csi_plugin.go`.
In fact, looking more closely at `pkg/volume/csi/` this whole package seems like an adaptation of the code from there.

Reply:
yes, I used pkg/volume/csi as an example for this code.
It doesn't mean that we can't make it better though. I can try to get rid of global variables.

btw, gochecknoglobals also warns about it:
```
pkg/kubelet/cm/dra/plugin/plugin.go:35:5: draPlugins is a global variable (gochecknoglobals)
var draPlugins = &PluginsStore{}
    ^
pkg/kubelet/cm/dra/plugin/plugin.go:40:5: PluginHandler is a global variable (gochecknoglobals)
var PluginHandler = &RegistrationHandler{}
```